### DISCUSSION TITLE
DEPTH_ZERO_SELF_SIGNED_CERT proxying to backend service

### DISCUSSION THREAD

**Discussion by robinbryce** [vercel/community#1260] [Help] [answered]
### Summary

I have a svelte based app, with a server side route that proxies to a backend service. The target service has a legitemate, production, tls certificate from lets encrypt. But all requests to it fail with 'DEPTH_ZERO_SELF_SIGNED_CERT'. Some searching suggested that the certificate is considered self signed but I don't see how.

Has anyone encountered this class of error before ?

### Additional information

The code that does the proxied request is this

`  // const options = deriveOptions(event.request)
  // const request = new Request(url, options)
  const request = new Request(proxyUrl, {
    method: event.request.method,
    body: JSON.stringify(data),
    keepalive: event.request.keepalive,
    headers: [
      ['content-type', 'application/json']
    ]
  })

  try {
    const upstream = await fetch(request)
    const body = JSON.stringify(await upstream.json())
    const response = new Response(body, {
      status: upstream.status,
      statusText: upstream.statusText,
      headers: [
        ['content-type', 'application/json'],
        ['date', upstream.headers.get('date')],
        ['vary', upstream.headers.get('vary')]
      ]
    })
    console.log('maptool status', response.statusText)
    return response
  } catch (err) {
    console.log('maptool headers', request.headers)
    console.log('fetch error:', err)
    return json(err)
  }`

**Comment by robinbryce** [comment, marked as answer]
Ah, curses, the env value was wrong and so hitting wrong sub domain. I swear I checked that but apprently not!

**Reply by amyegan** [reply]
Env variable mistakes trip us all from time to time. Glad you figured it out so quickly!
